File: data-structures/javascript/stacks/stack.js
class Stack {
  constructor() {
    this.items = [];
    this.size = 0;
  }

  insert(item) {
    this.items.push(item);
    this.size++;
  }

  pop() {
    this.size--;
    return this.items.pop();
  }

  contains(item) {
    return this.items.includes(item);
  }

  peek() {
    this.items[this.size - 1];
  }

  getMin() {
    return Math.min(...this.items);
  }

  getMax() {
    return Math.max(...this.items)
  }

  isEmpty() {
    return this.size === 0;
  }
}

const a = new Stack();
a.insert(10);
a.insert(9);
a.insert(11);
a.insert(120);
console.log("A IS: ", a);
a.pop();
console.log("NOW A IS: ", a);

File: data-structures/golang/stacks/stack.go
package stack

type Stack struct {
	Items [][]int
	Size  int
}

func InitStack() *Stack {
	var items = make([][]int, 0)
	return &Stack{
		Items: items,
		Size:  0,
	}
}

func (s *Stack) Insert(item int) error {
	s.items = append(s.items, item)
	s.size++
	return nil
}

func (s *Stack) Pop() (string, error) {
	removed := s.items[len(s.items)-1]
	s.items = s.items[0 : len(s.items)-2]
	return removed, nil
}
